tests/NHSD.BuyingCatalogue.Ordering.Api.UnitTests/Services/CreateOrderItemServiceTests.cs

```diff
@@ -21,5 +35,243 @@ public static void Constructors_VerifyGuardClauses()
 
             assertion.Verify(constructors);
         }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Constructor_NullDbContext_ThrowsArgumentNullException(
+            OrderItemValidator orderItemValidator,
+            ServiceRecipientService serviceRecipientService)
+        {
+            Assert.Throws<ArgumentNullException>(() => _ = new CreateOrderItemService(null, orderItemValidator, serviceRecipientService));
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Constructor_NullOrderItemValidator_ThrowsArgumentNullException(
+            ApplicationDbContext context,
+            ServiceRecipientService serviceRecipientService)
+        {
+            Assert.Throws<ArgumentNullException>(() => _ = new CreateOrderItemService(context, null, serviceRecipientService));
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static void Constructor_NullServiceRecipient_ThrowsArgumentNullException(
+            ApplicationDbContext context,
+            OrderItemValidator orderItemValidator)
+        {
+            Assert.Throws<ArgumentNullException>(() => _ = new CreateOrderItemService(context, orderItemValidator, null));
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static async Task CreateAsync_InvalidAggregateValidation_ValidateResult(
+            [Frozen] Mock<ICreateOrderItemValidator> orderItemValidator,
+            Order order,
+            ErrorDetails error,
+            CatalogueItemId catalogueItemId,
+            CreateOrderItemModel model,
+            CreateOrderItemService service)
+        {
+            var aggregateValidationResult = new AggregateValidationResult();
+            aggregateValidationResult.AddValidationResult(new ValidationResult(error), 0);
+
+            var catalogueItemType = Enum.Parse<CatalogueItemType>(model.CatalogueItemType, true);
+
+            orderItemValidator.Setup(o => o.Validate(order, model, catalogueItemType)).Returns(aggregateValidationResult);
+
+            var result = await service.CreateAsync(order, catalogueItemId, model);
+            result.Success.Should().BeFalse();
+        }
+
+        [Test]
+        [InMemoryDbAutoData]
+        public static async Task CreateAsync_ValidAggregateValidation_ValidateResult(
```

Comment:
As above, I think something like `CreateAsync_ValidAggregateValidation_SuccessIsTrue` might be a slightly better name.

Reply:
Renamed test method to `CreateAsync_ValidAggregateValidation_SuccessIsTrue `